docs/guides/blocks/formulas.md

```diff
@@ -30,20 +32,74 @@ Formulas can accept parameters, which is the primary way for them to access data
+      description: "Gets a Todoist task by URL",
+      parameters: [
+        coda.makeParameter({
+          type: coda.ParameterType.String,
+          name: "url",
+          description: "The URL of the task",
+        }),
+      ],
+      // ...
+      execute: async function ([url], context) {
+        // ...
+      },
+    });
+    ```
+
+By default if a Pack defines an authentication method then all formulas will require a connected account. You can change this behavior using the [`connectionRequirement`][connectionRequirement] property of the formula definition. Setting it to `Optional` makes the user account optional, and `None` removes the account parameter completely.
+
+While it's possible to use multiple accounts within a document, each instance of the formula must use a specific account. It isn't possible to have a formula that uses the account of whichever user is currently viewing the document. It's important for collaboration that all users in a Coda doc have access to the same data.
 
 
 ## Caching
 
-TODO
+For performance reasons all formula results are cached by default. If your formula is called again with the same parameters, the result will be loaded from the cache instead of re-running your code. Building or releasing a new version of your Pack will invalidate this cache, ensuring you get fresh results using your new code.
+
+You can adjust the caching behavior by setting the [`cacheTtlSecs`][cacheTtlSecs] field on the formula definition, which specifies for how many seconds the result should be cached. To disable caching for a formula set `cacheTtlSecs` to zero.
+
+!!! info
+    In addition to caching formula results, Coda also caches the [fetcher responses][fetcher_cache]. To get truly fresh results you may need to disable that caching as well.
 
 
 ## Recalculation
 
-TODO
+Coda has complex logic that determines when formula results need to be recalculated. Formulas will always recalculate when the parameter values change, but it will also happen periodically for other reasons including routine maintenance. In general you shouldn't make any assumptions about how often your formula will be run. If running your formula is expensive in some way (API cost, etc) make sure to use [caching](#caching) and [rate limits][fetcher_rate_limits] to limit the impact of recalculation.
+
+Building or releasing a new version of your Pack doesn't automatically cause existing formulas to recalculate, so users may still see old results for a while. In the **Settings** tab of the Pack's side panel there is a "Refresh now" button (**⟳**) that allows users to recalculate all formulas using the Pack. Pack makers also have access to the **Auto-refresh when version changes** toggle, which will do this automatically for the current window for each new Pack version.
+
+<img src="../../../images/settings_recalc.png" srcset="../../../images/settings_recalc_2x.png 2x" class="screenshot" alt="Recalculation settings in the Pack side panel">
```

Comment:
Just a reminder that these settings will all move into the new debug drawer in a week or two so we'll want to replace the screenshot at that time.

Reply:
Ah, good to know. I'll keep an eye out.